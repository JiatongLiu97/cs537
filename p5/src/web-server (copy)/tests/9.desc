Verify that shared memory is properly created in /dev/shm and that the size is PAGESIZE
