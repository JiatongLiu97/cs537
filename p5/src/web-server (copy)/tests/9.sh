#!/bin/bash

rm -f "/dev/shm/$TEST_SHM_NAME"

./server "$TEST_PORT" 1 1 "$TEST_SHM_NAME" > /dev/null 2>&1 &
server_pid="$!"

die() {
  kill -s SIGINT "$server_pid"
  exit 1
}

sleep 0.2

if [ -f "/dev/shm/$TEST_SHM_NAME" ]; then
  echo "Pass: shared memory successfully created in /dev/shm"
else
  echo "Fail: shared memory not created in /dev/shm"
  die
fi

shm_size="$(ls -l "/dev/shm/$TEST_SHM_NAME" | awk '{print $5}')"
page_size="$(getconf PAGESIZE)"
if [ "$shm_size" == "$page_size" ]; then
  echo "Pass: shared memory is the correct size"
else
  echo "Fail: shared memory is the wrong size"
  echo "Shared memory size: $shm_size"
  echo "Page size: $page_size"
  die
fi

kill -s SIGINT "$server_pid"
