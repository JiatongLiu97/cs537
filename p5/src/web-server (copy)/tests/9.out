Pass: shared memory successfully created in /dev/shm
Pass: shared memory is the correct size
